test/rules/trailingCommaRuleTests.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describe("<trailing-comma>", () => {
+    const TrailingCommaRule = Lint.Test.getRule("trailing-comma");
+    const fileName = "rules/trailingcomma.test.ts";
+
+    describe("multiline", () => {
+        it("restricts the use of trailing commas if set to never", () => {
+            const options = [true, {multiline: "never"}];
+
+            const expectedFailure1 = Lint.Test.createFailure(fileName, [6, 2], [6, 3], TrailingCommaRule.FAILURE_STRING_NEVER);
+            const expectedFailure2 = Lint.Test.createFailure(fileName, [5, 15], [5, 16], TrailingCommaRule.FAILURE_STRING_NEVER);
+            const expectedFailure3 = Lint.Test.createFailure(fileName, [16, 11], [16, 12], TrailingCommaRule.FAILURE_STRING_NEVER);
+            const expectedFailure4 = Lint.Test.createFailure(fileName, [24, 7], [24, 8], TrailingCommaRule.FAILURE_STRING_NEVER);
+            const expectedFailure5 = Lint.Test.createFailure(fileName, [45, 7], [45, 8], TrailingCommaRule.FAILURE_STRING_NEVER);
+            const expectedFailure6 = Lint.Test.createFailure(fileName, [55, 8], [55, 9], TrailingCommaRule.FAILURE_STRING_NEVER);
+            const expectedFailure7 = Lint.Test.createFailure(fileName, [73, 11], [73, 12], TrailingCommaRule.FAILURE_STRING_NEVER);
+            const actualFailures = Lint.Test.applyRuleOnFile(fileName, TrailingCommaRule, options);
+
+            Lint.Test.assertFailuresEqual(actualFailures, [
+              expectedFailure1,
+              expectedFailure2,
+              expectedFailure3,
+              expectedFailure4,
+              expectedFailure5,
+              expectedFailure6,
+              expectedFailure7
```

Comment:
4-space indentation please (sorry this isn't enforced; see https://github.com/palantir/tslint/issues/581)